collector/cloudwatch/settings.php

```diff
@@ -75,6 +75,12 @@
                 get_string('environment_desc', 'cltr_cloudwatch'),
                 'Dev', PARAM_TEXT));
         } else {
+            // When tool_cloudmetrics is installed, the AWS cloudwatch plugin is installed and enabled.
+            // Disable it if this is the first visit to the settings page and it has never been configured.
+            if (! get_config('cltr_cloudwatch', 'firstvisit')) {
```

Comment:
Most of the collectors we might eventually ship with should start out as disabled. ie if we support azure, prometheus, GCP, AWS, most people are only going to want one and just turn on that one.

So I think we want either:

option a) a way for each collector plugin to declare if it should default on
option b) we assume they should all default off, except database which is the special case and can default on
option c) we default them all off

I think the root cause here is that if there is no config at all then it assumes it is on:

```
$ php admin/cli/cfg.php --component=cltr_database
version	2022040500
metric_expiry	2592000
```

then if you disable it it is set:

```
$ php admin/cli/cfg.php --component=cltr_database
version	2022040500
metric_expiry	2592000
disabled	1
```
then re-enable:

```
$ php admin/cli/cfg.php --component=cltr_database
version	2022040500
metric_expiry	2592000
disabled	0
```

I think the simplest is to swap this logic so it must be explicitly enabled, and they default off, and we can enable the database either in install.php or upgrade.php

https://docs.moodle.org/dev/Plugin_files#db.2Finstall.php

Reply:
I do like that better.